pysatMadrigal/instruments/methods/general.py

```diff
@@ -481,13 +483,239 @@ def madrigal_file_format_str(inst_code, strict=False, verbose=True):
         if strict:
             raise ValueError(msg)
         elif verbose:
-            logger.warning(msg)
+            pysat.logger.warning(msg)
 
     fstr += "{file_type}"
 
     return fstr
 
 
+def sort_file_formats(fnames):
+    """Separate filenames by file format type.
+
+    Parameters
+    ----------
+    fnames : array-like
+        Iterable of filename strings, full path, to data files to be loaded.
+        This input is nominally provided by pysat itself.
+
+    Returns
+    -------
+    load_file_types : dict
+        A dictionary with file types as keys and a list of filenames for
+        each file type.
+
+    """
+    # Sort the files by file format type
+    load_file_types = {ftype: [] for ftype in file_types.keys()}
+    for fname in fnames:
+        for ftype in file_types.keys():
+            if fname.find(ftype) > 0:
+                load_file_types[ftype].append(fname)
+                break
+
+        if fname not in load_file_types[ftype]:
+            # Raise a logger warning if a file with an unknown extension
+            # is encountered
+            pysat.logger.warning(
```

Comment:
I wonder if this should be a warnings.warning instead. For your consideration since users can do something about this situation.

Reply:
I'd be fine with that on a future version, since we'd need to add a test.  Maybe make an issue?